poi/src/main/java/org/apache/poi/hssf/usermodel/HSSFWorkbook.java

```diff
@@ -297,11 +298,15 @@ public static String getWorkbookDirEntryName(DirectoryNode directory) {
                     "It must be decrypted before use by XSSF, it cannot be used by HSSF");
         }
 
-        // check for previous version of file format
-        if (directory.hasEntry(OLD_WORKBOOK_DIR_ENTRY_NAME)) {
```

Comment:
Not a big deal if this is too hard but is there any chance that `hasEntry` could remain case sensitive for backward compatibility reasons? And a new `caseInsensitiveHasEntry` added?

If leaving `hasEntry` case sensitive is too hard, could you could not using `directory.getEntryNames().contains` and adding a `caseSensitiveHasEntry` instead?

Reply:
Y, I don't like the backwards compatibility break that this proposed change brings about.  If we don't do this, though, then we'll have to change `hasEntry(name)` and `getEntry(name)` to the case insensitive options throughout our codebase.  I can do that if desired.

Yep, I'll add that now.